src/AutoRest.CSharp/Mgmt/Generation/MgmtClientBaseWriter.cs

```diff
@@ -605,44 +607,52 @@ protected void WritePagingMethodBranch(CSharpType itemType, Diagnostic diagnosti
         {
             var pagingMethod = operation.PagingMethod!;
             var firstPageRequestArguments = GetArguments(_writer, parameterMappings);
-            var nextPageRequestArguments = firstPageRequestArguments.IsEmpty() ? $"{KnownParameters.NextLink.Name}" : $"{KnownParameters.NextLink.Name}, {firstPageRequestArguments}";
+            firstPageRequestArguments.RemoveAt(firstPageRequestArguments.Count - 1);
 
-            FormattableString firstPageRequest = $"{GetRestClientName(operation)}.Create{pagingMethod.Method.Name}Request({firstPageRequestArguments})";
-            FormattableString? nextPageRequest = pagingMethod.NextPageMethod != null ? $"{GetRestClientName(operation)}.Create{pagingMethod.NextPageMethod.Name}Request({nextPageRequestArguments})" : (FormattableString?)null;
-            var pipelineReference = new Reference("Pipeline", typeof(HttpPipeline));
+            var restClient = new FormattableStringToExpression($"{GetRestClientName(operation)}");
+
+            _writer.WriteMethodBodyStatement(DeclareFirstPageRequestLocalFunction(restClient, pagingMethod.CreateRequestMethodName, firstPageRequestArguments, out var firstPageRequest));
+            CodeWriterDeclaration? nextPageRequest = null;
+            if (pagingMethod.CreateNextPageRequestMethodName is {} nextPageMethod)
+            {
+                var nextPageRequestArguments = firstPageRequestArguments.Prepend(KnownParameters.NextLink);
+                _writer.WriteMethodBodyStatement(DeclareNextPageRequestLocalFunction(restClient, nextPageMethod, nextPageRequestArguments, out nextPageRequest));
+            }
+
+            var clientDiagnostics = new FormattableStringToExpression($"{clientDiagnosticsReference.Name}");
+            var pipeline = new FormattableStringToExpression($"Pipeline");
             var scopeName = diagnostic.ScopeName;
             var itemName = pagingMethod.ItemName;
             var nextLinkName = pagingMethod.NextLinkName;
-
-            _writer.WritePageableBody(parameterMappings.Select(p => p.Parameter).Append(KnownParameters.CancellationTokenParameter).ToList(), itemType, firstPageRequest, nextPageRequest, clientDiagnosticsReference, pipelineReference, scopeName, itemName, nextLinkName, async);
+            _writer.WriteMethodBodyStatement(Return(CreatePageable(firstPageRequest, nextPageRequest, clientDiagnostics, pipeline, itemType, scopeName, itemName, nextLinkName, KnownParameters.CancellationTokenParameter, async)));
```

Comment:
I believe potentially the body of a method should be constructed in the output types instead of constructing in the writers, is it?
Could you please add a TODO here or somewhere else or create an issue to track this item of refactor?

Reply:
This is the last part of consolidation: https://github.com/Azure/autorest.csharp/issues/3772